src/HoverOverlay.tsx

```diff
@@ -126,96 +105,40 @@ export const HoverOverlay: <C>(props: HoverOverlayProps<C>) => React.ReactElemen
-                                                    <AlertCircleOutlineIcon className="icon-inline" /> Error rendering
-                                                    hover content
-                                                </strong>{' '}
-                                                {upperFirst(asError(err).message)}
-                                            </div>
-                                        )
-                                    }
-                                }
-                                return content.value
-                            }
-                            return (
-                                <code
-                                    className="hover-overlay__content hover-overlay__row e2e-tooltip-content"
-                                    key={i}
-                                    dangerouslySetInnerHTML={{
-                                        __html: highlightCodeSafe(content.value, content.language),
-                                    }}
-                                />
-                            )
-                        })
-                )}
-            </div>
-        )}
-
-        <div className="hover-overlay__actions hover-overlay__row">
-            <ButtonOrLink
-                linkComponent={linkComponent}
-                to={isJumpURL(definitionURLOrError) ? definitionURLOrError.jumpURL : undefined}
-                className="btn btn-secondary hover-overlay__action e2e-tooltip-j2d"
-                onClick={onGoToDefinitionClick ? transformMouseEvent(onGoToDefinitionClick) : undefined}
-            >
-                Go to definition {definitionURLOrError === LOADING && <LoadingSpinner className="icon-inline" />}
-            </ButtonOrLink>
-            {hoveredToken &&
-                referencesURL && (
-                    <ButtonOrLink
-                        linkComponent={linkComponent}
-                        // tslint:disable-next-line:jsx-no-lambda
-                        to={referencesURL}
-                        className="btn btn-secondary hover-overlay__action e2e-tooltip-find-refs"
-                    >
-                        Find references
-                    </ButtonOrLink>
-                )}
+        <div className="hover-overlay__contents">
+            {hoverOrError === LOADING ? (
+                <div className="hover-overlay__row hover-overlay__loader-row">
+                    <LoadingSpinner className="icon-inline" />
+                </div>
+            ) : isErrorLike(hoverOrError) ? (
+                <div className="hover-overlay__row hover-overlay__hover-error alert alert-danger">
+                    <h4>
+                        <AlertCircleOutlineIcon className="icon-inline" /> Error fetching hover from language server:
+                    </h4>
+                    {upperFirst(hoverOrError.message)}
+                </div>
+            ) : (
+                children
```

Comment:
We don’t use HoverOverlay in Sourcegraph, which is probably why this component feels weird now. It’s too abstract and not really useful. I would propose (see more in my comment below) removing HoverOverlay and extracting it to a separate package in the future.

Reply:
I don't understand why we don't use it anymore, it looks like it's now customisable enough?

Reply:
To be able to use it, we would need to add more abstraction here. It is already too abstract to be useful as is. It would not help other consumers use it, and it would make Sourcegraph’s code more complex unnecessarily.

I think there is a big tension here between making this library useful to potential future consumers and making it useful to Sourcegraph. I’d like to prioritize the latter and do what’s right for us to be able to use it in the way we need. Targeting potential consumers can come later.

Reply:
If we don't use it in Sourcegraph anymore, why are there changes other than converting it to use the provided actions instead of refs/def?

Reply:
I didn't want to spend the time updating it given that it won't be used standalone for at least a while (re: our discussion about that not being a priority). I am going to remove this HoverOverlay.tsx file to avoid confusion and unnecessary work. When/if we have a use case for this component outside of Sourcegraph, we can always add it back.